Add tags field to Alloydb Backup for TagsR2401

SamanthaMathews opened the pull request:
Add tags field to backup resource to allow setting tags on backup resources at creation time.
b/337048391
Release Note Template for Downstream PRs (will be copied)
```
alloydb: added `tags` field to `alloydb_backup` to allow setting tags for backups at creation time
```

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-11629-old..auto-pr-11629) ( 4 files changed, 113 insertions(+), 2 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-11629-old..auto-pr-11629) ( 4 files changed, 113 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-11629-old..auto-pr-11629) ( 1 file changed, 17 insertions(+))



## Missing test report
Your PR includes resource fields which are not covered by any test.

Resource: `google_alloydb_backup` (10 total tests)
Please add an acceptance test which includes these fields. The test should include the following:

```hcl
resource "google_alloydb_backup" "primary" {
  tags = # value needed
}


```

modular-magician commented:
#### Tests analytics
Total tests: 0
Passed tests: 0
Skipped tests: 0
Affected tests: 0

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>alloydb</li>

</ul>

</blockquote>
</details>
#### Non-exercised tests

Tests were added that are skipped in VCR:
- TestAccAlloydbBackup_tags
$\textcolor{red}{\textsf{Errors occurred during REPLAYING mode. Please fix them to complete your PR.}}$

View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11629/artifacts/3386cc74-88a5-4bff-a52d-41e822d7d188/build-log/replaying_test.log)

ScottSuarez reviewed (changes requested):
Hi Samantha! Thanks for contributing, three things:

1. The code is failing to compile. Here is the error from the logs
```log
# github.com/hashicorp/terraform-provider-google-beta/google-beta/services/alloydb
google-beta/services/alloydb/resource_alloydb_backup_test.go:326:2: expected declaration, found '}'
```

2. I don't see these fields documented on the API docs. Is that correct?

https://cloud.google.com/alloydb/docs/reference/rest/v1/projects.locations.backups

3. I don't appear to have access to the bug linked in the Pull Request - b/337048391

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report

Your PR generated some diffs in downstreams - here they are.

`google` provider: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-11629-old..auto-pr-11629) ( 4 files changed, 114 insertions(+), 2 deletions(-))
`google-beta` provider: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-11629-old..auto-pr-11629) ( 4 files changed, 114 insertions(+), 2 deletions(-))
`terraform-google-conversion`: [Diff](https://github.com/modular-magician/terraform-google-conversion/compare/auto-pr-11629-old..auto-pr-11629) ( 1 file changed, 17 insertions(+))

modular-magician commented:
#### Tests analytics
Total tests: 74
Passed tests: 69
Skipped tests: 4
Affected tests: 1

<details>
<summary>Click here to see the affected service packages</summary>
<blockquote>

<ul>
<li>alloydb</li>

</ul>

</blockquote>
</details>

#### Action taken
<details>
<summary>Found 1 affected test(s) by replaying old test recordings. Starting RECORDING based on the most recent commit. Click here to see the affected tests
</summary>
<blockquote>
<ul>
<li>TestAccAlloydbBackup_tags</li>

</ul>
</blockquote>
</details>

[Get to know how VCR tests work](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/contributing/#general-contributing-steps)

modular-magician commented:
$\textcolor{red}{\textsf{Tests failed during RECORDING mode:}}$
`TestAccAlloydbBackup_tags`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11629/artifacts/c8327771-5a01-47c3-943a-95f34d8235d1/build-log/recording_build/TestAccAlloydbBackup_tags_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11629/artifacts/c8327771-5a01-47c3-943a-95f34d8235d1/recording/TestAccAlloydbBackup_tags.log)]


$\textcolor{red}{\textsf{Errors occurred during RECORDING mode. Please fix them to complete your PR.}}$


View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-11629/artifacts/c8327771-5a01-47c3-943a-95f34d8235d1/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-11629/artifacts/c8327771-5a01-47c3-943a-95f34d8235d1/recording) for each test

c2thorn committed:
Convert from Ruby to Go

terraform-alloydb-backup-tags
commit terraform-alloydbbackup-tags
commit terraform-alloydb-backup-tags
commit terraform-alloydb-backup-tags
commit terraform-alloydb-backup-tags
